code/modules/food_and_drinks/drinks/drinks.dm

```diff
@@ -22,22 +22,28 @@
 		gulp_size = max(round(reagents.total_volume / 5), 5)
 
 /obj/item/reagent_containers/food/drinks/attack(mob/living/M, mob/user, def_zone)
-
 	if(!reagents || !reagents.total_volume)
-		to_chat(user, "<span class='warning'>[src] is empty!</span>")
-		return 0
+		if(iscyborg(user))
+			synthetize_reagents() //how the fuck do I call this
+		else
+			to_chat(user, "<span class='warning'>[src] is empty!</span>")
+		return FALSE
 
 	if(!canconsume(M, user))
-		return 0
+		return FALSE
 
 	if (!is_drainable())
 		to_chat(user, "<span class='warning'>[src]'s lid hasn't been opened!</span>")
-		return 0
+		return FALSE
 
-	if(M == user)
-		user.visible_message("<span class='notice'>[user] swallows a gulp of [src].</span>", "<span class='notice'>You swallow a gulp of [src].</span>")
-		if(M.has_trait(TRAIT_VORACIOUS))
-			M.changeNext_move(CLICK_CD_MELEE * 0.5) //chug! chug! chug!
+	if(M == user || iscyborg(user)) //Because M doesn't include silicons I guess?
```

Comment:
I'll try to compile this later. Cant now.

Reply:
Just a heads up, I joined a game and became a cyborg, and when I tried to use the cup in myself it did nothing. Maybe this is not the right way to call it, but this would technically allow you to start your analyzing by using the cup in another borg.

Also if a human try to feed a borg it will instead drink the liquids. While it is funny, I will try to fix this.